api/v1alpha1/metricsaggregate_types.go

```diff
@@ -0,0 +1,55 @@
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&MetricsAggregate{}, &MetricsAggregateList{})
+}
+
+// MetricsAggregateList contains a list of [MetricsAggregate]
+// +kubebuilder:object:root=true
+type MetricsAggregateList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []MetricsAggregate `json:"items"`
+}
+
+// MetricsAggregate
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:subresource:status
+type MetricsAggregate struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec of the MetricsAggregate
+	// +kubebuilder:validation:Required
+	Spec MetricsAggregateSpec `json:"spec"`
+
+	// Status of the IngressReplica
+	// +kubebuilder:validation:Optional
+	Status Status `json:"status,omitempty"`
+}
+
+type MetricsAggregateSpec struct {
+	Nodes int `json:"nodes"`
```

Comment:
are these fields optional?  Feels like this should probably be the status struct tbh

Reply:
I've moved to status